lms/product/d2l/_plugin/misc.py

```diff
@@ -0,0 +1,52 @@
+from lms.product.plugin.misc import MiscPlugin
+from lms.services.lti_grading.interface import LTIGradingService
+
+
+class D2LMiscPlugin(MiscPlugin):
+    def __init__(self, create_line_item: bool):
+        self._create_line_item = create_line_item
+
+    def post_configure_assignment(self, request):
+        """
+        Run any actions needed after configuring an assignment.
+
+        D2L doesn't create the container that holds grades (line item) LTI 1.3
+        assignments.
+
+        As a work-around, as soon as we configure the assignment on our end
+        we'll use the LTIA grading API to create a new line item we can use
+        later to record the grade.
+        """
+        lti_params = request.lti_params
+        lti_grading_service = request.find_service(LTIGradingService)
+
+        if not self._create_line_item:
+            # No need to do anything if this option is off.
+            return
+
+        resource_link_id = lti_params.get("resource_link_id")
+        resource_link_title = lti_params.get("resource_link_title")
+
+        # If we already identified this assignment as gradable no need to
+        # create anything
+        if not super().is_assignment_gradable(lti_params):
+            lti_grading_service.create_line_item(resource_link_id, resource_link_title)
```

Comment:
```suggestion
        # Nothing to do if line item creation is off, or it's already there
        if not self._create_line_item or super().is_assignment_gradable(
            request.lti_params
        ):
            return

        request.find_service(LTIGradingService).create_line_item(
            resource_link_id=request.lti_params.get("resource_link_id"),
            label=request.lti_params.get("resource_link_title"),
        )
```

I think we might have some left overs from refactors here?